model_analyzer/result/sorted_results.py

```diff
@@ -91,37 +98,66 @@ def top_n_results(self, n):
 
         Returns
         -------
-        list of ModelResults
+        list of RunConfigResults
             The n best results for this model,
             must all be passing results
         """
-
         if len(self._passing_results) == 0:
             logger.warning(
                 f"Requested top {n} configs, but none satisfied constraints. "
                 "Showing available constraint failing configs for this model.")
 
             if n == -1:
-                return heapq.nsmallest(len(self._failing_results),
-                                       self._failing_results)
+                return self._failing_results
```

Comment:
What is this special -1 case?

Reply:
See the comment in the method description. I'll boy scout this to remove the magic number and replace it with a constant

Reply:
Is it even possible to hit the -1 case in live code? I suppose it is safer to leave it in just in case.

However, it needs to be fully tested then. Our unit test coverage implies that the -1 case is covered for passing, but not for failing. Can you add a case with only failing tests with -1?

Reply:
I've added a case that exercises the failing only case.